### PR TITLE
Alternate host for private INTERNET network (like China)

### PR DESCRIPTION
**Pull request by Prometee**
Related issue #45 

This PR add a new `host` configuration value, allowing to change the used JS script host into the twig template.
It also change the host used to verify the captcha value.

### CODE REVIEW

**Review by karser — CHANGES_REQUESTED** [review]
Thank you for the PR. I'd like to suggest including `www.` in the host. And fix tests, please.

**Comment by karser** [comment]
**Error: Call to undefined method Karser\Recaptcha3Bundle\Tests\FunctionalTest::assertStringContainsString()**
Our composer.json: `"php": ">=7.1"` and we need assertStringContainsString in our tests.

phpunit v6.5.14 requires ["php": "^7.0"](https://github.com/sebastianbergmann/phpunit/blob/6.5.14/composer.json#L24) but it doesn't have `assertStringContainsString`. Only [assertContains](https://github.com/sebastianbergmann/phpunit/blob/6.5.14/src/Framework/Assert.php#L174)

phpunit v7.5.19 requires ["php": "^7.1"](https://github.com/sebastianbergmann/phpunit/blob/7.5.19/composer.json) and it supports [assertStringContainsString](https://github.com/sebastianbergmann/phpunit/blob/7.5.19/src/Framework/Assert.php#L2070)

phpunit  v8.5.20  requires ["php": ">=7.2"](https://github.com/sebastianbergmann/phpunit/blob/8.5.20/composer.json#L24), and it supports [assertStringContainsString](https://github.com/sebastianbergmann/phpunit/blob/8.5.20/src/Framework/Assert.php#L2672)


It should use phpunit v7.5.19 instead of v6.5.14.
Let's change `"phpunit/phpunit": "^6|^8|^9"` to `^7|^8|^9`

**Comment by karser** [comment]
**symfony/form[v5.0.0, ..., v5.0.8] require php ^7.2.5 -> your php version (7.1.33) does not satisfy that requirement.**

it seems we should avoid testing Symfony 5 with PHP 7.1

**Comment by Prometee** [comment]
Yes, I will take care of it, I already make exclusion in other repositories.
Can you let the GitHub Actions build on my push, it will avoid checking it until I fix the build. I will force push some commit to avoid a too long list.

**Comment by karser** [comment]
Done. Now 7.2 failed as well

**Comment by karser** [comment]
Error: "./composer.json" does not contain valid JSON
Parse error on line 59:
...0.1.x-dev"    },  }}

**Comment by Prometee** [comment]
Are you able to let the build auto start on my push ?

File: Tests/FunctionalTest.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function testHyphenConvertedToUnderscore()
 
         //THEN
         self::assertStringContainsString('<input type="hidden" id="form_capt-cha" name="form[capt-cha]" />', $view);
-        self::assertStringContainsString('<script type="text/javascript" src="https://www.google.com/recaptcha/api.js?render=key&onload=recaptchaCallback_form_capt_cha" async defer nonce=""></script>', $view);
+        self::assertStringContainsString('<script type="text/javascript" src="https://www.google.com/recaptcha/api.js?render=key&onload=recaptchaCallback_form_capt_cha" async defer></script>', $view);
```
[the remark this answers is not in the document]
Comment (Prometee) [reply Tests/FunctionalTest.php]: Yes, but the empty attribute have to be present if there is no nonce or an empty one ?

**Comment by karser** [comment]
> Are you able to let the build auto start on my push ?

I'm not sure. Do you know how to do that?

File: README.md [hunk 2]
```diff
@@ -75,6 +76,17 @@ RECAPTCHA3_SECRET=my_secret
...
 
+### How to use reCAPTCHA globally (meaning even in China):
+
+Use `'recaptcha.net'` host in your code when `'google.com'` is not accessible.
+
+```yaml
+# config/packages/karser_recaptcha3.yaml (or app/config/config.yml if using Symfony3)
+
+karser_recaptcha3:
+    host: 'recaptcha.net' # default is 'google.com'
```
Comment (karser) [inline README.md:87]: Should be `host: 'www.recaptcha.net' # default is 'www.google.com'`

File: README.md [hunk 3]
```diff
@@ -75,6 +76,17 @@ RECAPTCHA3_SECRET=my_secret
 Usage
 -----
 
+### How to use reCAPTCHA globally (meaning even in China):
+
+Use `'recaptcha.net'` host in your code when `'google.com'` is not accessible.
```
Comment (karser) [inline README.md:81]: Should be www.recaptcha.net / www.google.com

**Review by karser — CHANGES_REQUESTED** [review]

**Comment by Prometee** [comment]
> > Are you able to let the build auto start on my push ?
> 
> I'm not sure. Do you know how to do that?

It should be in the settings of this repository > Actions : https://github.com/karser/KarserRecaptcha3Bundle/settings/actions
Choose "Require approval for first-time contributors who are new to GitHub" instead of the second option.
it will be less annoying for you is more practical for those who create PR.

File: README.md (hunk 2 above)
Comment (Prometee) [reply README.md]: Yes, I forgot it 😉

**Comment by karser** [comment]
> > > Are you able to let the build auto start on my push ?
> > 
> > 
> > I'm not sure. Do you know how to do that?
> 
> It should be in the settings of this repository > Actions : https://github.com/karser/KarserRecaptcha3Bundle/settings/actions
> Choose "Require approval for first-time contributors who are new to GitHub" instead of the second option.
> it will be less annoying for you is more practical for those who create PR.

thanks, done!

**Comment by karser** [comment]
Looks like all is fine. I'm going to merge this and release it. Thank you @Prometee for the work on this PR!

**Comment by Prometee** [comment]
Thank you too for creating this project, it help us a lot 😉 !

**Comment by Prometee** [comment]
@karser re-reading the README.md I missed to add www. to the yaml comment ...
Can you add it quickly before releasing it ?

**Comment by karser** [comment]
Thanks, I noticed it too. Fixed it and moved a bit lower.

**Comment by Prometee** [comment]
Ok, perfect !

**Pull request merged**